crates/bevy_transform/src/systems.rs

```diff
@@ -236,6 +236,18 @@ unsafe fn propagate_to_child_unchecked(
     // Safety: `Children` is guaranteed to hold unique entities.
     #[expect(unsafe_code, reason = "Mutating disjoint entities in parallel")]
     let children_iter = unsafe { UniqueEntityIter::from_iterator_unchecked(children.iter()) };
+    // Performance note: iter_many tests every child to see if it meets the query. For leaf nodes,
```

Comment:
Is there a way you can add a filter to the query like `Without<IsParent>` to filter out the leaves?

Reply:
That filter is already present. The problem is that we have a list of entities from `Children` that need to be tested against that query.